src/util/Helpers.hpp

```diff
@@ -37,4 +39,28 @@ QColor getRandomColor(const QString &userId);
 QString formatUserMention(const QString &userName, bool isFirstWord,
                           bool mentionUsersWithComma);
 
+template <typename T>
+std::vector<T> splitListIntoBatches(const T &list, int batchSize = 100)
+{
+    std::vector<T> batches;
+    int batchCount = std::ceil(list.size() / batchSize);
```

Comment:
```suggestion
    int batchCount = std::ceil(static_cast<double>(list.size()) / batchSize);
```

When not using floating point numbers here, this breaks and never rounds up.